src/MQTTClient.php

```diff
@@ -143,29 +155,114 @@ public function connect(
      */
     protected function establishSocketConnection(): void
     {
-        if ($this->hasCertificateAuthorityFile()) {
-            $this->logger->info(sprintf('Using certificate authority file [%s] to verify peer name.', $this->caFile));
-
-            $socketContext = stream_context_create([
-                'ssl' => [
-                'verify_peer_name' => true,
-                    'cafile' => $this->getCertificateAuthorityFile(),
-                ],
-            ]);
-            $connectionString = 'tls://' . $this->getHost() . ':' . $this->getPort();
-            $this->socket     = stream_socket_client($connectionString, $errorCode, $errorMessage, 60, STREAM_CLIENT_CONNECT, $socketContext);
-        } else {
-            $connectionString = 'tcp://' . $this->getHost() . ':' . $this->getPort();
-            $this->socket     = stream_socket_client($connectionString, $errorCode, $errorMessage, 60, STREAM_CLIENT_CONNECT);
+        $useTls = ($this->settings->shouldUseTls() || $this->hasCertificateAuthorityFile());
```

Comment:
I guess you did this to avoid breaking backwards compatibility? For sure I'll change that in the upcoming v1.0. The CA file should not silently enable TLS in my opinion.

Reply:
This is exactly how it was in 0.1, guess 0.2 should preserve this behaviour?

Reply:
Yes, just wanted to make sure that was your intention as well. Let's keep it as that and change it later.